Source/WebCore/platform/graphics/skia/FontPlatformDataSkia.cpp

```diff
@@ -54,6 +54,45 @@ FontPlatformData::FontPlatformData(sk_sp<SkTypeface>&& typeface, float size, boo
     m_features = WTFMove(features);
 }
 
+FontPlatformData::FontPlatformData(float size, FontOrientation&& orientation, FontWidthVariant&& widthVariant, TextRenderingMode&& textRenderingMode, bool syntheticBold, bool syntheticOblique, RefPtr<FontCustomPlatformData>&& customPlatformData)
+    : FontPlatformData(size, syntheticBold, syntheticOblique, orientation, widthVariant, textRenderingMode, customPlatformData.get())
+{
+    m_font = SkFont(customPlatformData->m_typeface, m_size);
+    m_font.setEmbolden(m_syntheticBold);
+    m_font.setSkewX(m_syntheticOblique ? -SK_Scalar1 / 4 : 0);
+    m_font.setEdging(FontRenderOptions::singleton().antialias());
+    m_font.setHinting(FontRenderOptions::singleton().hinting());
+
+    m_hbFont = SkiaHarfBuzzFont::getOrCreate(*m_font.getTypeface());
+}
+
+std::optional<FontPlatformData> FontPlatformData::fromIPCData(float size, FontOrientation&& orientation, FontWidthVariant&& widthVariant, TextRenderingMode&& textRenderingMode, bool syntheticBold, bool syntheticOblique, IPCData&& ipcData)
+{
+    return WTF::switchOn(ipcData,
+        [&] (const FontPlatformSerializedData& d) -> std::optional<FontPlatformData> {
+            if (sk_sp<SkTypeface> typeface = SkTypeface::MakeDeserialize(SkMemoryStream::Make(d.typefaceData).get(), nullptr))
+                return FontPlatformData(WTFMove(typeface), size, syntheticBold, syntheticOblique, WTFMove(orientation), WTFMove(widthVariant), WTFMove(textRenderingMode), { });
+
+            return std::nullopt;
+        },
+        [&] (FontPlatformSerializedCreationData& d) -> std::optional<FontPlatformData> {
+            auto fontFaceData = SharedBuffer::create(WTFMove(d.fontFaceData));
+            if (RefPtr fontCustomPlatformData = FontCustomPlatformData::create(fontFaceData, d.itemInCollection))
+                return FontPlatformData(size, WTFMove(orientation), WTFMove(widthVariant), WTFMove(textRenderingMode), syntheticBold, syntheticOblique, WTFMove(fontCustomPlatformData));
```

Comment:
I guess we can use the existing constructor here, getting the typeface from the custom data and also passing the features that should be serialized too, I guess.